features/0335-http-over-didcomm/README.md

```diff
@@ -0,0 +1,165 @@
+- The **request-agent**: the aries agent which receives the HTTP request from the *requester*, and passes it along over DIDComm to the *response-agent*, and returns the reply in the other direction.
+- The **response-agent**: the aries agent which passes the request from the *request-agent* to the *responder*, receives the HTTP response, and passes it back along over DIDComm.
+- The **responder**: the entity (server) which receives the HTTP request sent by the *response-agent* on behalf of the *requester*, and replies with the response.
+
+When a *requester* sends an HTTP request to a *request-agent*, the request-agent may need to store a record of the requester's identity/IP, so the future response to this request can be sent to its intended destination.
+
+The *request-agent* can include some logic to decide whether to send the message, and may need to include some logic to decide where to route the message (note that in some architectures, another agent along the route makes the decision, so the request-agent might always send to the same target). If it does, it constructs a request DIDComm message (defined below) and sends it to the response-agent.
+
+The route taken by the DIDComm message across mediators, relays, etc, is not covered by this RFC.
+
+The *response-agent* receives the request DIDComm message. It can include some logic to decide whether to permit the message to continue to the *responder*. If so, it makes an HTTP request using the data in the request DIDComm message, and sends it to the responder.
+
+Note: in some use-cases, it might make sense for the response-agent to act as a transparent proxy, so the responder thinks its talking directly to the requester, while in others it might make sense to override requester identity information so the responder thinks it's connecting to the response-agent, or to a generic non-identifying interface. In this case, the request-agent could anonymize the request, rather than leaving it up to the response-agent.
+
+This same anonymization can be done in the other direction as well.
+
+The communication happens in reverse when the responder sends an HTTP response to the response-agent, which may again decide whether to permit it to continue. If so, the contents of the HTTP response are encoded into a response DIDComm message (defined below), sent to the request-agent, which also makes a go/no-go decision, does some logic (for example, looking up its thread-id to requester database) to figure out where the original request in this thread came from, encodes the response data into an HTTP response, and sends that response to the requester.
+
+#### Message Format
+This protocol defines new DIDComm message types to carry HTTP/1.x requests and responses.
+
+##### `request`
+
+```jsonc
+{
+  "@type": "https://didcomm.org/http-over-didcomm/1.0/request",
+  "@id": "2a0ec6db-471d-42ed-84ee-f9544db9da4b",
+  "request": {
+    "method": "",
+    "target": "",
+    "version": "",
+    "headers": {},
```

Comment:
Suggest this be made into an array. [Message headers](https://tools.ietf.org/html/rfc2616#section-4.2) are allowed to appear more than once, with a little bit of parsing logic to collapse them into a single unit. It seems wiser to just let the existing, battle-hardened http components handle this.